src/main/cpp/_nix_based/jssc.cpp

```diff
@@ -723,6 +723,9 @@ JNIEXPORT jobjectArray JNICALL Java_jssc_SerialNativeInterface_waitEvents
     jclass intClass = env->FindClass("[I");
     jobjectArray returnArray = env->NewObjectArray(sizeof(events)/sizeof(jint), intClass, NULL);
 
+    struct pollfd waitingSet = { static_cast<int>(portHandle), POLLIN, 0 };
+    poll(&waitingSet, 1, 1000);
```

Comment:
`poll` seem to have [bad portability](https://cr.yp.to/docs/unixport.html) for some unixes, please add a precompiler check to ensure this feature is available.

Reply:
This page seems incredibly old, it seems poll was added in 2005 to Mac OS (though it was sort of [broken](https://daniel.haxx.se/blog/2016/10/11/poll-on-mac-10-12-is-broken/) for some time). I am not sure how to specifically check for `poll` availability with precompiler checks, so the best I could think of is to check whether we're on Mac

Reply:
Agreed.  I'd be ok with a 10.9+ requirement, but 10.12+ is probably too new.  We can revisit this solution for Mac at a later time.